projects/igniteui-angular/src/lib/test-utils/grid-base-components.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { ColumnDefinitions, GridTemplateStrings } from './template-strings.spec';
+
+@Component({
+    template: `
+        <igx-grid
+            [data]="data">
+        </igx-grid>
+    `
+})
+export class BasicGridComponent {
+    @Input()
+    public data = [];
+
+    @ViewChild(IgxGridComponent)
+    public grid: IgxGridComponent;
+}
+
+@Component({
+    template: `
+        <igx-grid
+            [data]="data"
+            [autoGenerate]="autoGenerate">
+        </igx-grid>
+    `
+})
+export class GridAutoGenerateComponent extends BasicGridComponent {
+    @Input()
+    public autoGenerate = true;
+}
+
+@Component({
+    template: `
+        <igx-grid
+            [data]="data"
+            [autoGenerate]="autoGenerate"
+            [height]="height" [width]="width">
+        </igx-grid>
+    `
+})
+export class GridWithSizeComponent extends GridAutoGenerateComponent {
+    @Input()
+    public width: string;
+
+    @Input()
+    public height: string;
+}
+
+@Component({
+    template: `
+        <igx-grid
+            [data]="data"
+            [autoGenerate]="autoGenerate"
+            [height]="height" [width]="width">
+        </igx-grid>
+    `
+})
+export class GridNxMComponent extends GridWithSizeComponent implements OnInit {
+    public colsCount: number;
+    public rowsCount: number;
+    public columnsType = 'string';
+    public hasEditableColumns = false;
+    public startFromOne = false;
+    public columnNamePrefix = 'col';
+    public columns = [];
+
+    ngOnInit() {
+        this.columns = (this.hasEditableColumns) ? SampleTestData.generateEditableColumns(this.colsCount, this.columnNamePrefix)
+                                : SampleTestData.generateColumnsByType(this.colsCount, this.columnsType, this.columnNamePrefix);
+        this.data = SampleTestData.generateDataForColumns(this.columns, this.rowsCount, this.startFromOne);
+    }
+
+    public isHorizonatScrollbarVisible() {
+        const scrollbar = this.grid.parentVirtDir.getHorizontalScroll();
+        return scrollbar.offsetWidth < scrollbar.children[0].offsetWidth;
+    }
+}
+
+@Component({
+    template: GridTemplateStrings.declareBasicGridWithColumns(ColumnDefinitions.idNameJobHireDate)
+})
+export class BasicGridSearchComponent extends BasicGridComponent {
+    public highlightClass = 'igx-highlight';
+    public activeClass = 'igx-highlight__active';
+}
+
+@Component({
+    template: GridTemplateStrings.declareGrid(` [paging]="paging" [perPage]="perPage"`,
+        '', ColumnDefinitions.idNameJobTitle)
+})
+export class PagingComponent extends BasicGridComponent {
+    public paging = true;
+    public perPage = 3;
+
+    data = SampleTestData.personJobData;
+}
+
+@Component({
+    template: GridTemplateStrings.declareGrid(` [rowSelectable]="rowSelectable"`,
+        '', ColumnDefinitions.productBasic)
+})
+export class SelectionComponent extends BasicGridComponent {
+    public rowSelectable = true;
```

Comment:
Due to consistency I think here we should have default passed data.

Reply:
Done!